monkey/tests/unit_tests/monkey_island/cc/repository/test_mongo_event_repository.py

```diff
@@ -0,0 +1,177 @@
+import uuid
+from typing import List
+from unittest.mock import MagicMock
+
+import mongomock
+import pytest
+from pydantic import Field
+
+from common.event_serializers import EventSerializerRegistry, PydanticEventSerializer
+from common.events import AbstractAgentEvent
+from monkey_island.cc.repository import (
+    IEventRepository,
+    MongoEventRepository,
+    RemovalError,
+    RetrievalError,
+    StorageError,
+)
+
+
+class FakeAgentEvent(AbstractAgentEvent):
+    data = Field(default=435)
+
+
+class FakeAgentItemEvent(AbstractAgentEvent):
+    item: str
+
+
+EVENTS: List[AbstractAgentEvent] = [
+    FakeAgentEvent(source=uuid.uuid4(), tags={"foo"}),
+    FakeAgentEvent(source=uuid.uuid4(), tags={"foo", "bar"}),
+    FakeAgentEvent(source=uuid.uuid4(), tags={"bar", "baz"}),
+    FakeAgentItemEvent(source=uuid.uuid4(), tags={"baz"}, item="blah"),
+]
+
+
+@pytest.fixture
+def event_serializer_registry() -> EventSerializerRegistry:
+    registry = EventSerializerRegistry()
+    registry[FakeAgentEvent] = PydanticEventSerializer(FakeAgentEvent)
+    registry[FakeAgentItemEvent] = PydanticEventSerializer(FakeAgentItemEvent)
+    return registry
+
+
+@pytest.fixture
+def mongo_client(event_serializer_registry):
+    client = mongomock.MongoClient()
+    client.monkey_island.events.insert_many(
+        (event_serializer_registry[type(e)].serialize(e) for e in EVENTS)
+    )
+    return client
+
+
+@pytest.fixture
+def mongo_repository(mongo_client, event_serializer_registry) -> IEventRepository:
+    return MongoEventRepository(mongo_client, event_serializer_registry)
+
+
+@pytest.fixture
+def error_raising_mongo_client(mongo_client) -> mongomock.MongoClient:
+    mongo_client = MagicMock(spec=mongomock.MongoClient)
+    mongo_client.monkey_island = MagicMock(spec=mongomock.Database)
+    mongo_client.monkey_island.events = MagicMock(spec=mongomock.Collection)
+
+    # The first call to find() must succeed
```

Comment:
Is this true or just copy/paste?

Reply:
Not true, just a residual comment from borrowed code